websites/views.py

```diff
@@ -252,3 +249,9 @@ def get_serializer_context(self):
         if site_config and config_item is not None:
             added_context["is_page_content"] = is_page_content(site_config, config_item)
         return {**super().get_serializer_context(), **added_context}
+
+    def perform_destroy(self, instance: WebsiteContent):
+        """ (soft) deletes a WebsiteContent record """
+        super().perform_destroy(instance)
+        update_website_backend(instance.website)
```

Comment:
`api.update_website_backend` calls `backends.sync_website_content` which currently doesn't handle deletes.  Should it?  Or change this to `api.sync_content` -> `backend.sync_content_to_backend` -> `backend.delete_content_in_backend`?

```python
sync_content(instance.content_sync_state)
```

Reply:
I'm going to let you merge your changes first, since you modified `upsert_content_files_for_user`